src/main/java/com/bisnode/opa/process/OpaTestProcess.java

```diff
@@ -24,16 +32,34 @@ public ProcessExecutionResult execute() {
                     .directory(rootDir)
                     .command(command.getCommandArgs())
                     .start();
+
+            Future<String> resultFromOpa = spawnOutputConsumerThread(process);
             int exitCode = process.waitFor();
-            return new ProcessExecutionResult(asString(process.getInputStream()), exitCode);
-        } catch (IOException | InterruptedException e) {
+            // Should only retrieve results after OPA process exits
+            String result = resultFromOpa.get();
+            return new ProcessExecutionResult(result, exitCode);
+        } catch (IOException | InterruptedException | ExecutionException e) {
             throw new TestExecutionException("Failed to start OPA process for tests", e);
         }
     }
 
-    private String asString(InputStream inputStream) {
-        // "Stupid Scanner trick" https://community.oracle.com/blogs/pat/2004/10/23/stupid-scanner-tricks
-        return new Scanner(inputStream, UTF_8.name()).useDelimiter("\\A").next();
+    private Future<String> spawnOutputConsumerThread(Process process) {
```

Comment:
How about extracting this to a separate class, since we do the same in `StartOpaTask`?